app/src/main/java/com/mediaan/mymediaan/view/ftu/CreateProfileScreen.kt

```diff
@@ -0,0 +1,291 @@
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.rememberDrawerState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavController
+import com.mediaan.mymediaan.R
+import com.mediaan.mymediaan.model.TwoTruthsOneLieEntity
+import com.mediaan.mymediaan.repository.ProfileRepository
+import com.mediaan.mymediaan.ui.theme.MediaanPrimary
+import com.mediaan.mymediaan.ui.theme.Typography
+import com.mediaan.mymediaan.view.MyMediaanAppBar
+import com.mediaan.mymediaan.viewModel.MainNavigationViewModel
+import com.mediaan.mymediaan.viewModel.MyMediaanScreen
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CreateProfileScreen(
+    navController: NavController,
+    viewModel: MainNavigationViewModel,
+    profileRepository: ProfileRepository,
+) {
+    Scaffold(
+        topBar = {
+            MyMediaanAppBar(
+                drawerState = rememberDrawerState(initialValue = DrawerValue.Closed),
+                title = stringResource(MyMediaanScreen.Ftu.title),
+                navIcon = Icons.Filled.ArrowBack,
+                iconDescr = "Back",
+                onNavIconClick = { navController.popBackStack() }
+            )
+        },
+    ) { innerPadding ->
+        // TODO: should we this move logic to new view model - ftuViewModel?
+        val scrollState = rememberScrollState()
+        var firstName by remember { mutableStateOf("") }
+        var nickName by remember { mutableStateOf("") }
+        var lastName by remember { mutableStateOf("") }
+        val minCharName = 3
+        val maxCharName = 50
+        var age by remember { mutableStateOf("") }
+        val minAge = 18
+        val maxAge = 122
+        var selectedOffice by remember { mutableStateOf(viewModel.offices[0]) }
+        var isExpanded by remember { mutableStateOf(false) }
+        var firstTruth by remember { mutableStateOf("") }
+        var secondTruth by remember { mutableStateOf("") }
+        var lie by remember { mutableStateOf("") }
+        val minCharFacts = 10
+        val maxCharFacts = 280
+        val isProfileValid = remember {
+            derivedStateOf {
+                firstName.isNotEmpty() &&
+                        lastName.isNotEmpty() &&
+                        age.isNotEmpty() &&
+                        firstTruth.isNotEmpty() &&
+                        secondTruth.isNotEmpty() &&
+                        lie.isNotEmpty()
+            }
+        }.value
```

Comment:
@Inn0 what do you think, should this be moved to separate ViewModel or we can leave this code here?

Reply:
Personally I would have introduced a ViewModel, but since we're not using these values outside of this composable directly I think it is fine as-is. (so its up to you ;))

Reply:
At the end I decided to implement view model.